### PR TITLE
Generates Dataframe JSON with supporting ETL functions

### LINKED ISSUE #38: Species, Locations, Papers, Habitats, and Records DataFrames
Create the starting Pandas dataframes for the SPOC Verifier App

- [x] Using the WoRms Taxon csv at `SUL AI 2020-2021/Project - Species Occurrences/WoRMS_marineSpecies/taxon.txt` _species_ dataframe with three columns: 
   1. **taxonID**
   2. **scientificName**.
   3. **references** as URL
- [x] Using the three locations csv files `SUL AI 2020-2021/Project - Species Occurrences/data/NamedPlaces_CA.csv`, `SUL AI 2020-2021/Project - Species Occurrences/data/NamedPlaces_OR.csv`, and `SUL AI 2020-2021/Project - Species Occurrences/data/NamedPlaces_WA.csv`, create a _locations_ dataframe with the following columns: 
   1. **FEATURE_ID**
   2. **FEATURE_NAME**
   3. **STATE_ALPHA** as STATE
   4. **PRIM_LONG_DEC** as LONGITUDE
   5. **PRIM_LAT_DEC** as LATITUDE
- ~~From the converted TEI XML in `SUL AI 2020-2021/Project - Species Occurrences/papers_tei` and the Google metadata [Spreadsheet](https://docs.google.com/spreadsheets/d/1EjCjMBLPdELkdR-XSz31gv8jZwBOCwCZEG0vNlLj57I/edit?usp=drive_fs&ouid=102548702195264613391), create a _papers_ Dataframe with two columns~~:
  ~~1. **XML filename**~~
  ~~2. **Institution**~~ Moved to it's own issue, https://github.com/sul-dlss-labs/spoc/issues/47
- [x] Use the `SUL AI 2020-2021/Project - Species Occurrences/data/habitats.jsonl`, create a __habitats__ Dataframe with one column **habitat**
- [x] From the papers TEI XML, create a _records_ Dataframe that uses the four Dataframes above and from the full-text, create a _records_ dataframe with the following columns:  
  1. **Paper ID** (the name of TEI XML file)
  2. **Instance ID** (the taxonID) 
  3. **Species** (scientificName column)
  4.  **GBIF** URI search from the scientificName
  5. **Time** (pulled from TEI XML) 
  6. **Place** (list of FEATURE_NAMEs from _locations_ df)
  7. **Habitats** (from habitat df)
  8. **div_number** (the number (1-based)  index of the DIV from the paper)

All Dataframes should be saved in the SPOC data sub-directory as Dataframe JSON

### PR DESCRIPTION
**Pull request by jermnelson**
Fixes #38, created new `src/lib/etl.py` module for extracting data from our data sources, transforming into panda DataFrames, and then loading results after processing through `nlp` pipeline for the new `species-records.json` DataFrame that will be used by the SPOC Verifier Streamlit app.

### CODE REVIEW

File: src/lib/etl.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import spacy  # type: ignore
+import pandas as pd  # type: ignore
+import lxml.etree as etree  # type: ignore
+
+from typing import Any, Dict, List
+
+TEI = {"tei": "http://www.tei-c.org/ns/1.0"}
+
+
+def get_date(xml_doc: etree.Element) -> str:
```
Comment (jgreben) [inline src/lib/etl.py:11]: Does anything call this method (besides the test)?

File: src/lib/etl.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    # TODO: Check doc._.overlap and add to lists
+
+    # Removes duplicates from species, locations, and habitats
+    species = list(set(species))
+    locations = list(set(places))
+    habitats = list(set(habitats))
+    return species, locations, habitats
+
+
+def get_all_text(xml_doc: etree.Element) -> str:
```
Comment (jgreben) [inline src/lib/etl.py:51]: Does anything call this method (besides the test)?

File: src/lib/etl.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                "Time": publication_date,
+                "Place": locations,
+                "Habitats": habitats,
+                "div_enum": div_number,
+            }
+        )
+    return records
+
+
+def process_xml(raw_xml: str, filename: str, nlp: spacy.lang) -> list:
```
Comment (jgreben) [inline src/lib/etl.py:91]: Does anything call this?

**Review by jgreben — COMMENTED** [review]
Maybe a walkthrough would be in order. I don't see where some of the methods are being called. Are they called somewhere external to this repo?

File: src/lib/etl.py (hunk 1 above)
Comment (jermnelson) [reply src/lib/etl.py:11]: > Maybe a walkthrough would be in order. I don't see where some of the methods are being called. Are they called somewhere external to this repo?

@jgreben  sounds good; maybe pair after stand-up?

File: src/lib/etl.py (hunk 3 above)
Comment (jgreben) [reply src/lib/etl.py:91]: Ok, I see it now in the data-prep notebook,

File: src/lib/etl.py (hunk 1 above)
Comment (jgreben) [reply src/lib/etl.py:11]: My apologies, I missed where this called in your walkthrough and I do not see it in data-prep. Can you just add the comment to where it is called here. Thanks!

File: src/lib/etl.py (hunk 2 above)
Comment (jermnelson) [reply src/lib/etl.py]: No, removed from latest commit.

File: src/lib/etl.py (hunk 1 above)
Comment (jermnelson) [reply src/lib/etl.py:11]: Done, added comment in `process_xml` function.

**Comment by jermnelson** [comment]
@jgreben removed function `get_all_text` along with accompanying tests. Also, added a comment to the `process_xml` function about being called from the data-prep notebook. Added new ticket #58  for testing `process_xml` and `process_div` functions.

**Review by jgreben — APPROVED** [review]

**Pull request merged**